### PR TITLE
add NAD CIDR validation to webhook

### PR DESCRIPTION
**Pull request by chrisho**
**Problem:**
https://github.com/harvester/harvester/issues/3580

**Solution:**
Add a CIDR validation to webhook

**Related Issue:**
https://github.com/harvester/harvester/issues/3580

**Test plan:**
https://github.com/harvester/harvester/issues/3580

### CODE REVIEW

File: pkg/utils/nad.go [hunk 1]
```diff
@@ -54,10 +54,10 @@ func NewLayer3NetworkConf(conf string) (*Layer3NetworkConf, error) {
 	if networkConf.Mode != "" && networkConf.Mode != Auto && networkConf.Mode != Manual {
 		return nil, fmt.Errorf("unknown mode %s", networkConf.Mode)
 	}
-	if _, _, err := net.ParseCIDR(networkConf.CIDR); networkConf.CIDR != "" && err != nil {
+	if _, _, err := net.ParseCIDR(networkConf.CIDR); networkConf.Mode == Manual && networkConf.CIDR != "" && err != nil {
```
Comment (guangbochen) [inline pkg/utils/nad.go:57]: IIRC, both DHCP and Manual mode would allow specifying the DHCP server.

Comment (chrisho) [reply pkg/utils/nad.go:57]: Actually not in UI, only specifying the DHCP server in auto mode.

**Review by bk201 — APPROVED** [review]
LGTM

File: pkg/utils/nad.go [hunk 2]
```diff
@@ -54,10 +54,10 @@ func NewLayer3NetworkConf(conf string) (*Layer3NetworkConf, error) {
...
 		return nil, fmt.Errorf("invalid CIDR %s", networkConf.CIDR)
 	}
-	if networkConf.Gateway != "" && net.ParseIP(networkConf.Gateway) == nil {
+	if networkConf.Mode != Manual && networkConf.Gateway != "" && net.ParseIP(networkConf.Gateway) == nil {
```
Comment (futuretea) [inline pkg/utils/nad.go:60]: networkConf.Mode == Manual

**Review by futuretea — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**